src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -16,64 +18,67 @@ static class NpgsqlActivitySource
-            // but then again mixes the concept of database and schema.
-            // As I interpret it, it actually wants db.sql.table to include the schema name and not the
-            // database name if the concept of schemas exists in the database system.
-            // This also makes sense in the context of the activity name which otherwise would include the
-            // database name twice.
-            dbSqlTable = commandText;
-            activityName = $"{dbOperation} {dbName}.{dbSqlTable}";
+            // We follow the {db.operation.name} {target} pattern of the spec, with the operation being SELECT and
+            // the target being the table (collection) name.
+            operationName = "SELECT";
+            spanName ??= $"{operationName} {commandText}";
             break;
         case CommandType.Text:
-            activityName = dbName;
+            // We don't have db.query.summary, db.operation.name or target (without parsing SQL),
+            // so we fall back to db.system.name as per the specs.
+            spanName ??= "postgresql";
             break;
         default:
             throw new ArgumentOutOfRangeException(nameof(commandType), commandType, null);
         }
 
-        var activity = Source.StartActivity(spanName ?? activityName, ActivityKind.Client);
+        var activity = Source.StartActivity(spanName, ActivityKind.Client);
         if (activity is not { IsAllDataRequested: true })
             return activity;
 
-        activity.SetTag("db.statement", commandText);
+        activity.SetTag("db.query.text", commandText);
 
-        if (dbOperation != null)
-            activity.SetTag("db.operation", dbOperation);
-        if (dbSqlTable != null)
-            activity.SetTag("db.sql.table", dbSqlTable);
+        switch (commandType)
+        {
+            case CommandType.StoredProcedure:
+                Debug.Assert(operationName is not null);
+                activity.SetTag("db.operation.name", operationName);
+                activity.SetTag("db.stored_procedure.name", commandText);
+                break;
+            case CommandType.TableDirect:
+                Debug.Assert(operationName is not null);
+                activity.SetTag("db.operation.name", operationName);
+                activity.SetTag("db.collection.name", commandText);
+                break;
+        }
 
         return activity;
     }
 
-    internal static Activity? ConnectionOpen(NpgsqlConnector connector)
+    internal static Activity? PhysicalConnectionOpen(NpgsqlConnector connector)
     {
         if (!connector.DataSource.Configuration.TracingOptions.EnablePhysicalOpenTracing)
             return null;
 
         var dbName = connector.Settings.Database ?? connector.InferredUserName;
-        var activity = Source.StartActivity(dbName, ActivityKind.Client);
+        var activity = Source.StartActivity("connect " + dbName, ActivityKind.Client);
```

Comment:
I think a good idea will be rename it to to `CONNECT` (the spec does say to use `BATCH`, `SELECT` etc, so we probably should follow the convention) and maybe add a comment that it's not in the spec but it's better compared to having the exact same name as for queries